src/plugins/sockets/inGameEvents.js

```diff
@@ -25,9 +25,9 @@ export async function handleInGameEvents(evData) {
   }
   // Handle GameOver
   if (evData.victory && evData.victory.gameOver) {
-    setTimeout(() => {
+    sleep(1000).then(() => {
       gameStore.setGameOver(evData.victory);
-    }, 1000);
+    });
```

Comment:
I tried to use async await here like I did in the other parts of the PR but for some reason, it does not work unless it's promise chained

Reply:
@seriouslysean any idea why tf `.then()` works but `await` doesn't? It looks to be the case that the subsequent lines fire at the appropriate time either way, but when using `await` the DOM fails to update.

Reply:
Created #923 to follow up

Reply:
Lemme dig in to this a bit today and see if I can suss anything out.

Reply:
Just to be sure, you were doing something like this?

```
// Handle GameOver
if (evData.victory?.gameOver) {
  await sleep(1000);
  gameStore.setGameOver(evData.victory);
}
```

Reply:
Yea

Reply:
@Haviles04 any chance you can get a minimal reproduction of it not working?

Reply:
@seriouslysean I'm at work ATM but can do it either tonight or tomorrow